icx-proxy/src/config/canister_dns_config.rs

```diff
@@ -143,6 +146,86 @@ mod tests {
         )
     }
 
+    #[test]
+    fn searches_longest_to_shortest() {
+        // If we checked these in the order passed, a.b.c would erroneously resolve
+        // to the canister id associated with b.c
+        let dns_aliases = parse_dns_aliases(vec![
+            "b.c:rrkah-fqaaa-aaaaa-aaaaq-cai",
+            "a.b.c:r7inp-6aaaa-aaaaa-aaabq-cai",
+        ])
+        .unwrap();
+
+        assert_eq!(
+            dns_aliases.resolve_canister_id_from_split_hostname(&["a", "b", "c"]),
+            Some(Principal::from_text("r7inp-6aaaa-aaaaa-aaabq-cai").unwrap())
+        );
+        assert_eq!(
+            dns_aliases.resolve_canister_id_from_split_hostname(&["d", "b", "c"]),
+            Some(Principal::from_text("rrkah-fqaaa-aaaaa-aaaaq-cai").unwrap())
+        );
+    }
+
+    #[test]
+    fn searches_longest_to_shortest_even_if_already_ordered() {
+        // Similar to searches_longest_to_shortest, just to ensure that
+        // we do the right thing no matter which order they are passed.
+        let dns_aliases = parse_dns_aliases(vec![
+            "a.b.c:r7inp-6aaaa-aaaaa-aaabq-cai",
+            "b.c:rrkah-fqaaa-aaaaa-aaaaq-cai",
+        ])
+        .unwrap();
+
+        assert_eq!(
+            dns_aliases.resolve_canister_id_from_split_hostname(&["a", "b", "c"]),
+            Some(Principal::from_text("r7inp-6aaaa-aaaaa-aaabq-cai").unwrap())
+        );
+        assert_eq!(
+            dns_aliases.resolve_canister_id_from_split_hostname(&["d", "b", "c"]),
+            Some(Principal::from_text("rrkah-fqaaa-aaaaa-aaaaq-cai").unwrap())
+        );
+    }
+
+    #[test]
+    fn searches_longest_to_shortest_not_alpha() {
+        // Similar to searches_longest_to_shortest, but make sure we
+        // don't happen to get there by sorting alphabetically
+        let dns_aliases = parse_dns_aliases(vec![
+            "a.x.c:r7inp-6aaaa-aaaaa-aaabq-cai",
+            "x.c:rrkah-fqaaa-aaaaa-aaaaq-cai",
+        ])
+        .unwrap();
+
+        assert_eq!(
+            dns_aliases.resolve_canister_id_from_split_hostname(&["a", "x", "c"]),
+            Some(Principal::from_text("r7inp-6aaaa-aaaaa-aaabq-cai").unwrap())
+        );
+        assert_eq!(
+            dns_aliases.resolve_canister_id_from_split_hostname(&["d", "x", "c"]),
+            Some(Principal::from_text("rrkah-fqaaa-aaaaa-aaaaq-cai").unwrap())
+        );
+    }
+
+    #[test]
+    fn searches_longest_to_shortest_not_alpha_reversed() {
+        // Similar to searches_longest_to_shortest, but make sure we
+        // don't happen to get there by sorting alphabetically/reversed
+        let dns_aliases = parse_dns_aliases(vec![
+            "x.a.c:r7inp-6aaaa-aaaaa-aaabq-cai",
+            "a.c:rrkah-fqaaa-aaaaa-aaaaq-cai",
```

Comment:
Can you reverse the order you pass them? If it's not sorted this test would still pass, right?

Reply:
The test above passes them in the reversed order from this one.  Does that cover the case you were wanting to?

Reply:
Yeah, I missed something on this one.. will adjust